internal/config/dynamic_config_watcher.go

```diff
@@ -31,12 +35,11 @@ func NewDynamicConfigWatcher(canaryConfig *CanaryConfig, applyFunc func(config *
 			applyFunc(target)
 		}
 
-		dynamicConfigWatcher.ticker = time.NewTicker(canaryConfig.DynamicConfigWatcherInterval * time.Millisecond)
-		quit := make(chan struct{})
 		go func() {
 			for {
+				ticker := time.NewTicker(canaryConfig.DynamicConfigWatcherInterval * time.Millisecond)
```

Comment:
shouldn't this ticker creation be out of the `for` loop (still inside the go routine of course)? It sounds like every time you get a ticker event and the corresponding code runs in the select, then back to the for loop a new ticker is created.
It sounds to be wrong to me, am I missing something?

Reply:
you are right, that was my intent.